app/src/org/commcare/utils/CommCareUtil.java

```diff
@@ -82,8 +82,7 @@ public static int countPreds(TreeReference reference) {
     }
 
     public static void triggerLogSubmission(Context c) {
-        SharedPreferences settings = CommCareApplication.instance().getCurrentApp().getAppPreferences();
-        String url = settings.getString(CommCareServerPreferences.PREFS_SUBMISSION_URL_KEY, null);
+        String url = LogSubmissionTask.getSubmissionUrl(CommCareApplication.instance().getCurrentApp().getAppPreferences());
```

Comment:
@ctsims Can you elaborate on that? Definitely not something I currently avoid, would be good to understand why